### ISSUE TITLE
Signal 4.2.0 crashes when opening a group conversation

### ISSUE THREAD

**Issue by paride** [WhisperSystems/Signal-Android#6464]
### Steps to reproduce
- Update to 4.2.0
- Open a group conversation
- Signal crashes (always reproducible)

### Device info

**Device:** Virtual Nexus 5X, Nexus 6
**Android version:** 7.1.1
**Signal version:** 4.2.0

### Link to debug log

https://gist.github.com/anonymous/08f80fda756ce09926920f6cb2970095

**Comment by nrizzio** [comment]
```
--------- beginning of crash
03-31 21:07:03.911  2539  2539 E AndroidRuntime: FATAL EXCEPTION: main
03-31 21:07:03.911  2539  2539 E AndroidRuntime: Process: org.thoughtcrime.securesms, PID: 2539
03-31 21:07:03.911  2539  2539 E AndroidRuntime: java.lang.ClassCastException: android.support.v7.app.ActionBar$LayoutParams cannot be cast to android.widget.LinearLayout$LayoutParams
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.widget.LinearLayout.measureVertical(LinearLayout.java:731)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.widget.LinearLayout.onMeasure(LinearLayout.java:640)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.view.View.measure(View.java:19731)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.widget.RelativeLayout.measureChildHorizontal(RelativeLayout.java:715)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.widget.RelativeLayout.onMeasure(RelativeLayout.java:461)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.view.View.measure(View.java:19731)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6120)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1464)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.widget.LinearLayout.measureHorizontal(LinearLayout.java:1117)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.widget.LinearLayout.onMeasure(LinearLayout.java:642)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.view.View.measure(View.java:19731)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.support.v7.widget.RecyclerView$LayoutManager.measureChildWithMargins(RecyclerView.java:8456)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.support.v7.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1551)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.support.v7.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1488)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.support.v7.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:560)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.support.v7.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:3506)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.support.v7.widget.RecyclerView.dispatchLayout(RecyclerView.java:3254)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.support.v7.widget.RecyclerView.consumePendingUpdateOperations(RecyclerView.java:1593)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.support.v7.widget.RecyclerView$ViewFlinger.run(RecyclerView.java:4625)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:871)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.view.Choreographer.doCallbacks(Choreographer.java:683)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.view.Choreographer.doFrame(Choreographer.java:616)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:857)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:751)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:95)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:154)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6077)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:865)
03-31 21:07:03.911  2539  2539 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:755)
```

**Comment by cbarnsdale** [comment]
Same issue on the OnePlus 3. I can provide another log if needed.

**Comment by nrizzio** [comment]
This crash only happens when a message *you* send is in view. If you open a conversation where all the most recent messages are from other people, the app won't crash. But when scrolling up, the moment one of your own messages come into view, the app crashes.

**Comment by jjohns71** [comment]
I can confirm I am experiencing this bug when I go to open an MMS thread that I sent a message in on Nexus 6P running Android 7.1.2. 

[Link to my debug log](https://gist.github.com/anonymous/6e9bb733ce007f3cf2a81dc6f22885f2) where Signal crashed twice.  Hopefully this is helpful.

**Comment by riyapenn** [comment]
A couple other logs: 
1. https://gist.github.com/1d22ab8e1d746f4b0555f41b8495d730
2. https://gist.github.com/d92113ea2d84b27b462a3ce2cc67d464

**Comment by mrgtreeskier** [comment]
@moxie0 I still have this issue when opening group messages regardless of who sent the last message.

Signal 4.2.0.
Android 5.0.2.

Log: https://gist.github.com/anonymous/505a14d81b6d61c21ac5c9c413d7a4f1

**Comment by jjohns71** [comment]
@moxie0 thank you for the Friday Fix! How can I send some $ towards your project?

**Comment by haffenloher** [comment]
@mrgtreeskier the fix will be included in version 4.2.1.

@jjohns71 you can donate via https://freedom.press/crowdfunding/signal/ or contribute funds to OWS's Bithub instance (https://whispersystems.org/blog/bithub/) which rewards external contributors
